site/public/js/notebook_builder/widgets/item-widget.js

```diff
@@ -0,0 +1,135 @@
+class ItemWidget extends Widget {
+    constructor() {
+        super();
+
+        this.dom_pointer;
+
+        this.state = {
+            'type': 'item',
+            'from_pool': []
+        }
```

Comment:
Hii @Kevin-MM , Itempool support is working really cool but I think `user_item_map` feature is missing.

Reply:
Hey @mukul-kmr-jha , thanks for the review.  We hadn't discussed having that feature available in the notebook builder.  I think it might be a lower priority item to add that to the notebook builder at the moment.  Did you find any other issues in your review?

Reply:
Thanks @mukul-kmr-jha 
Yes, we will want the user_item_map feature soonish -- it's a great convenience for instructor debugging.
It can wait until a later PR -- but hopefully soon :)

Reply:
Sounds good to me...and everything else is working great, thanks :)